tests/Twilio/Unit/Rest/ClientTest.php

```diff
@@ -241,4 +242,21 @@ public function testAuthorizationHeaderRemoval(): void {
         $this->assertStringNotContainsString('Authorization-header', stream_get_contents($capturedLogging));
     }
 
+    public function testDefaultUserAgent(): void {
+        $client = new Client('username', 'password');
+        $client->request('GET', 'https://api.twilio.com');
+        $userAgent = $client->getHttpClient()->{'lastRequest'}[CURLOPT_HTTPHEADER][0];
+        $expectedString = 'twilio-php/' . VersionInfo::string() . ' (' . php_uname("s") . ' ' . php_uname("m") . ')' . ' PHP/' . PHP_VERSION;
+        $this->assertStringContainsString($expectedString,$userAgent);
+    }
+
+    public function testUserAgentExtensionsWhenSet(): void {
+        $userAgentExtensions = ['twilio-run/2.0.0-test', 'flex-plugin/3.4.0'];
+        $client = new Client('username', 'password', null,null,null,null,$userAgentExtensions);
+        $client->request('GET', 'https://api.twilio.com');
+        $userAgent = $client->getHttpClient()->{'lastRequest'}[CURLOPT_HTTPHEADER][0];
+        $expectedString = 'twilio-php/' . VersionInfo::string() . ' (' . php_uname("s") . ' ' . php_uname("m") . ')' . ' PHP/' . PHP_VERSION . ' ' . implode(' ',$userAgentExtensions);
+        $this->assertStringContainsString($expectedString,$userAgent);
```

Comment:
Should this assert on an exact match instead?

Reply:
$userAgent will show "User-Agent: twilio-php/6.36.0 (Darwin x86_64) PHP/8.1.4" due to the "key: value" concatenation in CurlClient. Should I do a split on the $userAgent string to get an exact match instead?

Reply:
Think do an exact match and include the key prefix in the match.

Reply:
Noted.